### PR TITLE
Group the subcommands of karmadactl

### PR DESCRIPTION
**Pull request by lonelyCZ**
Signed-off-by: lonelyCZ <531187475@qq.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
With increasement of subcommands that have different functions, perhaps, it is clearer to group them.

```
[root@master67 karmada]# ./karmadactl -h
karmadactl controls a Kubernetes Cluster Federation.

Basic Commands:
  get           Display one or many resources
  version       Print the version information

Cluster Registeration Commands:
  init          Install karmada in kubernetes
  deinit        Removes Karmada from Kubernetes
  join          Register a cluster to control plane
  unjoin        Remove the registration of a cluster from control plane

Cluster Management Commands:
  cordon        Mark cluster as unschedulable
  uncordon      Mark cluster as schedulable
  taint         Update the taints on one or more clusters

Troubleshooting and Debugging Commands:
  logs          Print the logs for a container in a pod in a cluster
  exec          Execute a command in a container in a cluster
  describe      Show details of a specific resource or group of resources in a cluster

Advanced Commands:
  apply         Apply a configuration to a resource by file name or stdin and propagate them into member clusters
  promote       Promote resources from legacy clusters to karmada control plane

Other Commands:
  completion    Generate the autocompletion script for the specified shell

Usage:
  karmadactl [flags] [options]

Use "karmadactl <command> --help" for more information about a given command.
```

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
As for how to group, we can have more discussion.

**Does this PR introduce a user-facing change?**:

```release-note
karmadactl: grouping the subcommands.
```

### CODE REVIEW

File: pkg/karmadactl/karmadactl.go [hunk 1]
```diff
@@ -46,20 +47,53 @@ func NewKarmadaCtlCommand(cmdUse, parentCommand string) *cobra.Command {
...
-	rootCmd.AddCommand(NewCmdDescribe(karmadaConfig, parentCommand))
-	rootCmd.AddCommand(cmdinit.NewCmdInit(parentCommand))
-	rootCmd.AddCommand(NewCmdDeInit(parentCommand))
+
+	groups := templates.CommandGroups{
+		{
+			Message: "Basic Commands:",
+			Commands: []*cobra.Command{
+				NewCmdGet(karmadaConfig, parentCommand),
+				sharedcommand.NewCmdVersion(parentCommand),
```
Comment (RainbowMango) [inline pkg/karmadactl/karmadactl.go:56]: move `version` to group `other`?

Comment (lonelyCZ) [reply pkg/karmadactl/karmadactl.go:56]: Agree!

Comment (lonelyCZ) [reply pkg/karmadactl/karmadactl.go]:
```
[root@master67 karmada]# ./karmadactl -h
karmadactl controls a Kubernetes Cluster Federation.

Basic Commands:
  get           Display one or many resources

Cluster Registeration Commands:
  init          Install karmada in kubernetes
  deinit        Removes Karmada from Kubernetes
  join          Register a cluster to control plane
  unjoin        Remove the registration of a cluster from control plane

Cluster Management Commands:
  cordon        Mark cluster as unschedulable
  uncordon      Mark cluster as schedulable
  taint         Update the taints on one or more clusters

Troubleshooting and Debugging Commands:
  logs          Print the logs for a container in a pod in a cluster
  exec          Execute a command in a container in a cluster
  describe      Show details of a specific resource or group of resources in a cluster

Advanced Commands:
  apply         Apply a configuration to a resource by file name or stdin and propagate them into member clusters
  promote       Promote resources from legacy clusters to karmada control plane

Other Commands:
  completion    Generate the autocompletion script for the specified shell
  version       Print the version information

Usage:
  karmadactl [flags] [options]
```

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**